packages/engine/lib/stateful/src/state/references.rs

```diff
@@ -1,3 +1,10 @@
+/// A record pointing to a specific [`Agent`] inside of an [`AgentPool`].
+#[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct AgentIndex {
+    pub agent_batch_index: u32,
+    pub agent_index: u32,
```

Comment:
Not sure how we want to clarify this but it's worth pointing out that this is the index _within_ the group, rather than the index in the complete State of Agents